src/Resources/ResourceManager/Utilities/ResourceIdParser.cs

```diff
@@ -0,0 +1,75 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Commands.ResourceManager.Cmdlets.Utilities
+{
+    using System.Text.RegularExpressions;
+
+    public static class ResourceIdParser
```

Comment:
>ResourceIdParser [](start = 24, length = 16)

I thought we have a utility class today. does that not work?

Reply:
Are you refereeing to `Microsoft.Azure.Commands.Common.Strategies.ResourceId`? I checked the implementation and it doesn't handle a special case where a resource group is considered as a subscription level resource and there's no provider in the ID. For example:

`/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/myRG`

The `ResourceId` TryParse will return null given such a resource ID, while in the what-if scenario we would like to extract the scope and resource group provider/name from it:
```
- /subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
- /Microsoft.Resources/resourceGroups/myRG
```

Reply:
cool